src/main/scala/scorex/core/ModifiersCache.scala

```diff
@@ -92,29 +84,28 @@ trait LRUCache[PMOD <: PersistentNodeViewModifier, HR <: HistoryReader[PMOD, _]]
   // complete scan and cleaning of removed keys happen.
   private val cleaningThreshold = 50
 
-  @tailrec
-  private def evictionCandidate(): K = {
-    val k = evictionQueue.dequeue()
-    if(cache.contains(k)) k else evictionCandidate()
-  }
 
   override protected def onPut(key: K): Unit = {
     evictionQueue.enqueue(key)
-    if(evictionQueue.size > maxSize + cleaningThreshold){
+    if (evictionQueue.lengthCompare(maxSize + cleaningThreshold) > 0) {
       evictionQueue.dequeueAll(k => !cache.contains(k))
     }
   }
 
-  override protected def onRemove(key: K, rememberKey: Boolean): Unit = {
-  }
+  def cleanOverfull(): Seq[V] = {
+    @tailrec
+    def removeUntilCorrectSize(acc: Seq[V]): Seq[V] = if (size <= maxSize || evictionQueue.isEmpty) {
+      acc
+    } else {
+      removeUntilCorrectSize(remove(evictionQueue.dequeue()).toSeq ++ acc)
```

Comment:
we can make `acc` a `List` and use `evictionQueue.dequeue()) :: acc` expression to prepend new elements without converting them to `Seq`